### PR TITLE
Resets all state generated by Keras in evaluator to avoid memory leak.

### PR DESCRIPTION
**Pull request by zhjunqin**
In distributed evaluator, it will do continuous evaluation. 
But TF keras keeps a global cache for each evaluation model graph.
If not clear after evary evaluation, it will lead to memory leak.

The problem not found in TF 1.14 version, becuase TF 1.x version auto release graph object after each evaluation. But TF 2.x changed such behavior.

please check tf.keras.backend.clear_session.

```
@keras_export('keras.backend.clear_session')
def clear_session():
  """Resets all state generated by Keras.

  Keras manages a global state, which it uses to implement the Functional
  model-building API and to uniquify autogenerated layer names.

  If you are creating many models in a loop, this global state will consume
  an increasing amount of memory over time, and you may want to clear it.
  Calling `clear_session()` releases the global state: this helps avoid clutter
  from old models and layers, especially when memory is limited.

  Example 1: calling `clear_session()` when creating models in a loop

  ``python
  for _ in range(100):
    # Without `clear_session()`, each iteration of this loop will
    # slightly increase the size of the global state managed by Keras
    model = tf.keras.Sequential([tf.keras.layers.Dense(10) for _ in range(10)])

  for _ in range(100):
    # With `clear_session()` called at the beginning,
    # Keras starts with a blank state at each iteration
    # and memory consumption is constant over time.
    tf.keras.backend.clear_session()
    model = tf.keras.Sequential([tf.keras.layers.Dense(10) for _ in range(10)])
  ``

  Example 2: resetting the layer name generation counter

  >>> import tensorflow as tf
  >>> layers = [tf.keras.layers.Dense(10) for _ in range(10)]
  >>> new_layer = tf.keras.layers.Dense(10)
  >>> print(new_layer.name)
  dense_10
  >>> tf.keras.backend.set_learning_phase(1)
  >>> print(tf.keras.backend.learning_phase())
  1
  >>> tf.keras.backend.clear_session()
  >>> new_layer = tf.keras.layers.Dense(10)
  >>> print(new_layer.name)
  dense
  """
  global _SESSION
  global _GRAPH_LEARNING_PHASES  # pylint: disable=global-variable-not-assigned
  global _GRAPH_VARIABLES  # pylint: disable=global-variable-not-assigned
  global _GRAPH_TF_OPTIMIZERS  # pylint: disable=global-variable-not-assigned
  global _GRAPH
  _GRAPH.graph = None
  ops.reset_default_graph()
  reset_uids()
  _SESSION.session = None
  graph = get_graph()
  with graph.as_default():
    _DUMMY_EAGER_GRAPH.learning_phase_is_set = False
    _GRAPH_LEARNING_PHASES.clear()
    # Create the learning phase placeholder in graph using the default factory.
    _GRAPH_LEARNING_PHASES.setdefault(graph)
    _GRAPH_VARIABLES.pop(graph, None)
    _GRAPH_TF_OPTIMIZERS.pop(graph, None)
```

### CODE REVIEW

**Comment by zhjunqin** [comment]
@googlebot I signed it!

File: tensorflow_estimator/python/estimator/training.py [hunk 1]
```diff
@@ -903,6 +903,10 @@ def _execute_evaluator_once(self, evaluator, continuous_eval_listener,
           'before finding (next) new checkpoint. Please consider to increase '
           'it.')
 
+    # Resets all state generated by Keras.
+    # If not do clear, it will lead to memory leak.
+    tf.keras.backend.clear_session()
```
Comment (qlzh727) [inline tensorflow_estimator/python/estimator/training.py:908]: I am not sure if it will cause any issue since this function is executed by _start_continuous_evaluation, and might incur some performance loss if the session is cleared and some cache might lost. Also since I am not the expert for estimator, we might need someone else to review this change.

**Comment by gbaned** [comment]
@zhjunqin  Can you please check @qlzh727's comments and keep us posted ? Thanks!

File: tensorflow_estimator/python/estimator/training.py (hunk 1 above)
Comment (zhjunqin) [reply tensorflow_estimator/python/estimator/training.py:908]: I see, I will close this PR.

**Comment by zhjunqin** [comment]
> @zhjunqin Can you please check @qlzh727's comments and keep us posted ? Thanks!

Thanks for reminder. I'm considering close this PR.
